### PR TITLE
test: test rewardBadges

### PR DESCRIPTION
**Pull request by igorntwari**
# Submit a pull request

- [ ] This is not a duplicate of an existing pull request.
- [ ] No existing features have been broken without good reason.
- [ ] Your commit messages are detailed
- [ ] The code style guideline have been followed.
- [ ] Documentation has been updated to reflect your changes.
- [ ] Tests have been added or updated to reflect your changes.
- [ ] All tests pass.

---

## Pull Request Details

Provide details about your pull request and what it adds, fixes, or changes.

## Issues Fixed

Enter the issue numbers resolved by this pull request below, if any.

#1106

### CODE REVIEW

File: src/components/badges/RewardBadge.tsx [hunk 1]
```diff
@@ -43,10 +43,10 @@ export default function RewardBadge({ reward = {}, type = "transparent", display
   });
   if (!reward) return <></>;
   return (
-    <span className={badgeClassnames}>
+    <span data-testid="RewardBadge" className={badgeClassnames}>
```
Comment (Samantha-KY) [inline src/components/badges/RewardBadge.tsx:46]: Can you add the test id as a prop

**Review by serapieTuyishime — APPROVED** [review]

File: __tests__/components/badges/RewardBadges.test.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+describe("RewardBadges", () => {
+  it("should render  rewardBadges", () => {
+    render(<RewardBadge />);
+    const rewardBadge = screen.getByTestId("RewardBadge");
+    expect(rewardBadge).toBeInTheDocument();
+  });
+
+  it("should render rewardBadges with coins", () => {
+    render(<RewardBadge />);
+    render(<Coin />);
```
Comment (mhenriette) [inline __tests__/components/badges/RewardBadges.test.tsx:15]: Avoid rendering the `Coin` component separately as it is already rendered within the `RewardBadge` component 
Something like:
```suggestion
      it("should render rewardBadges with Coin when the token is provided", () => {
    render(<RewardBadge reward={{ token: "BTC" }} />);
    const rewardBadge = screen.queryByTestId("RewardBadge");
    const coin = screen.getByTestId("coin");
    expect(rewardBadge).toBeInTheDocument();
    expect(coin).toBeInTheDocument();
  });
```

File: __tests__/components/badges/RewardBadges.test.tsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    expect(coin).toBeInTheDocument();
+  });
+  it("should render rewardBadges with amount and token", () => {
+    render(<RewardBadge reward={{ token: "BTC", amount: 1000 }} displayAmount={true} />);
+    const rewardBadge = screen.getByTestId("RewardBadge");
+    const coin = screen.getByTestId("coin");
+    const amount = screen.getByTestId("amount");
+    expect(rewardBadge).toBeInTheDocument();
+    expect(coin).toBeInTheDocument();
+    expect(amount).toBeInTheDocument();
```
Comment (mhenriette) [inline __tests__/components/badges/RewardBadges.test.tsx:28]: there's no need of testing this again since you tested it while rendering rewardBadges

File: __tests__/components/badges/RewardBadges.test.tsx [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    const rewardBadge = screen.queryByTestId("RewardBadge");
+    const coin = screen.getByTestId("coin");
+    expect(rewardBadge).toBeInTheDocument();
+    expect(coin).toBeInTheDocument();
+  });
+  it("should render rewardBadges with amount and token", () => {
+    render(<RewardBadge reward={{ token: "BTC", amount: 1000 }} displayAmount={true} />);
+    const rewardBadge = screen.getByTestId("RewardBadge");
+    const coin = screen.getByTestId("coin");
+    const amount = screen.getByTestId("amount");
```
Comment (mhenriette) [inline __tests__/components/badges/RewardBadges.test.tsx:25]: avoid using data-testid on every element you want to test, `data-testId` should be used as a last approach, to avoid changing the existing implementation only just for testing.

File: __tests__/components/badges/RewardBadges.test.tsx [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    render(<RewardBadge />);
+    render(<Coin />);
+    const rewardBadge = screen.queryByTestId("RewardBadge");
+    const coin = screen.getByTestId("coin");
+    expect(rewardBadge).toBeInTheDocument();
+    expect(coin).toBeInTheDocument();
+  });
+  it("should render rewardBadges with amount and token", () => {
+    render(<RewardBadge reward={{ token: "BTC", amount: 1000 }} displayAmount={true} />);
+    const rewardBadge = screen.getByTestId("RewardBadge");
```
Comment (mhenriette) [inline __tests__/components/badges/RewardBadges.test.tsx:23]: you can change this test like this to check the amount using `getByText`
```suggestion
const text = screen.getByText('shortened-1000 BTC')
    expect(text).toBeInTheDocument()
```
'shortened-1000 BTC' is returned by the mocked function 

`jest.mock("../../../src/utilities", () => ({
  shortenNumber: (num:number) => `shortened-${num}`
}))`

File: src/components/badges/RewardBadge.tsx [hunk 6]
```diff
@@ -43,10 +43,10 @@ export default function RewardBadge({ reward = {}, type = "transparent", display
...
       {token && <Coin token={token} size="small" />}
       {amount && (
-        <div className="font-medium pl-0 pr-2">
+        <div data-testid="amount" className="font-medium pl-0 pr-2">
```
Comment (mhenriette) [inline src/components/badges/RewardBadge.tsx:49]: remove this data-testid, it is no needed considering the above  reason.

**Review by mhenriette — CHANGES_REQUESTED** [review]
You should also add the test of checking what is displayed once the reward is empty

**Review by mhenriette — APPROVED** [review]

**Review by musayann — APPROVED** [review]
